java/core/src/main/java/com/microsoft/bond/Marshal.java

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.bond;
+
+import com.microsoft.bond.helpers.ArgumentHelper;
+import com.microsoft.bond.protocol.ProtocolWriter;
+
+import java.io.IOException;
+
+/**
+ * Resposible for marshaling of objects.
+ */
+public final class Marshal {
+
+    // prevent instantiation
+    private Marshal() {
+    }
+
+    /**
+     * Marshals a struct object using the given protocol writer.
+     *
+     * @param obj       object to marshal
+     * @param writer    protocol writer
+     * @param <TStruct> the type of the object
+     * @throws IOException if an I/O error occurred
+     */
+    public static <TStruct extends BondSerializable> void marshal(TStruct obj, ProtocolWriter writer)
+            throws IOException {
+        ArgumentHelper.ensureNotNull(obj, "obj");
+        ArgumentHelper.ensureNotNull(writer, "writer");
+        Serializer<TStruct> serializer = new Serializer<TStruct>();
```

Comment:
Diamond operator is supported by Java 7+. We are using 6.

---
In reply to: [134009984](https://github.com/Microsoft/bond/pull/571#discussion_r134009984) [](ancestors = 134009984)